packages/ketcher-macromolecules/src/components/monomerLibrary/RnaBuilder/RnaAccordion/RnaAccordion.tsx

```diff
@@ -91,11 +94,48 @@ export const RnaAccordion = ({ libraryName, duplicatePreset, editPreset }) => {
 
   const [expandedAccordion, setExpandedAccordion] =
     useState<RnaBuilderItem | null>(activeRnaBuilderItem);
+  const [newPreset, setNewPreset] = useState(activePreset);
 
+  useEffect(() => {
+    setNewPreset(activePreset);
+  }, [activePreset]);
+
+  const dispatch = useDispatch();
   const handleAccordionSummaryClick = (rnaBuilderItem: RnaBuilderItem) => {
     if (expandedAccordion === rnaBuilderItem) {
       setExpandedAccordion(null);
-    } else setExpandedAccordion(rnaBuilderItem);
+    } else {
+      dispatch(setActiveRnaBuilderItem(rnaBuilderItem));
```

Comment:
Is this logic needed here? I mean it doesn't look so straightforward to call it on accordion entry click for me

Reply:
Checked, we may remove this, it won't affect validations